src/coord/src/coord.rs

```diff
@@ -1006,13 +1043,20 @@ impl Coordinator {
 
     /// Instruct the dataflow layer to cancel any ongoing, interactive work for
     /// the named `conn_id`.
-    async fn handle_cancel(&mut self, conn_id: u32) {
-        // Tell dataflow to cancel any pending peeks.
-        self.broadcast(SequencedCommand::CancelPeek { conn_id });
+    async fn handle_cancel(&mut self, conn_id: u32, secret_key: u32) {
+        if let Some(conn_meta) = self.active_conns.get(&conn_id) {
+            // If the secret key specified by the client doesn't match the
+            // actual secret key for the target connection, we treat this as a
+            // rogue cancellation request and ignore it.
+            if conn_meta.secret_key == secret_key {
```

Comment:
`!=`?

Reply:
Oy, thanks! Luckily there was plenty of test coverage for this.